src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs

```diff
@@ -85,6 +89,45 @@ public ResultWrapper<GethLikeTxTrace> debug_traceCall(TransactionForRpc call, Bl
         if (_logger.IsTrace) _logger.Trace($"{nameof(debug_traceTransaction)} request {tx.Hash}, result: trace");
         return ResultWrapper<GethLikeTxTrace>.Success(transactionTrace);
     }
+    public ResultWrapper<IEnumerable<GethLikeTxTrace>> debug_traceCallMany(TransactionForRpcWithTraceTypes[] calls, BlockParameter? blockParameter = null)
+    {
+        blockParameter ??= BlockParameter.Latest;
+        using CancellationTokenSource cancellationTokenSource = new(_traceTimeout);
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
+
+        SearchResult<BlockHeader> headerSearch = _debugBridge.SearchBlockHeaderForTraceCall(blockParameter);
+        if (headerSearch.IsError)
+        {
+            return ResultWrapper<IEnumerable<GethLikeTxTrace>>.Fail(headerSearch);
+        }
+
+        if (!_debugBridge.HasStateForBlock(headerSearch.Object))
+        {
+            return ResultWrapper<IEnumerable<GethLikeTxTrace>>.Fail($"No state available for block {headerSearch.Object.ToString(BlockHeader.Format.FullHashAndNumber)}", ErrorCodes.ResourceUnavailable);
+        }
+
+        Dictionary<Hash256, ParityTraceTypes> traceTypeByTransaction = new(calls.Length);
+        Transaction[] txs = new Transaction[calls.Length];
+        for (int i = 0; i < calls.Length; i++)
+        {
+            calls[i].Transaction.EnsureDefaults(_jsonRpcConfig.GasCap);
+            Transaction tx = calls[i].Transaction.ToTransaction();
+            tx.Hash = new Hash256(new UInt256((ulong)i).ToBigEndian());
+            txs[i] = tx;
+        }
```

Comment:
All logic should live in `GethStyleTracer`

Reply:
Hey, I'm sorry but I didn't understand what you meant here? Did you mean using `GethStyleTracer` instead of `GethLikeTrace` as object?